src/shmem_node_util.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include "shmem_internal.h"
+#include "shmem_node_util.h"
+
+
+int *shmem_internal_location_array = NULL;
+
+static int node_util_is_initialized = 0;
+static char my_hostname[SHMEM_INTERNAL_MAX_HOSTNAME_LEN];
+static int n_local_pes = 0;
+
+int shmem_node_util_init(void)
+{
+    int ret;
+    char errmsg[256];
+
+    if (!node_util_is_initialized) {
+        if (gethostname(my_hostname, SHMEM_INTERNAL_MAX_HOSTNAME_LEN)) {
```

Comment:
I think what we might want to do is replace `gethostname` with `shmem_transport_gethostname` and then allow the transport layer to supply this info.  That should allow us to build a single table of hostnames in the PMI KVS (e.g. reuse the addresses we gather to build the AV/Portals map).

Reply:
The `gethostname` code would move to transport-none.  Hopefully for Portals and OFI we can use the addresses they already gather to populate address maps.